vertexai/multimodal-pdf/pdf.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// multimodalpdf shows an example of understanding a PDF file input
+package multimodalpdf
+
+// [START generativeaionvertexai_gemini_pdf]
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/vertexai/genai"
+)
+
+// generateContentFromPDF generates a response into w, based upon the prompt
+// and the pdf file provided.
+// pdf is a Google Cloud Storage path starting with "gs://"
+func generateContentFromPDF(w io.Writer, prompt, pdf, projectID, location, modelName string) error {
+	// prompt := `
+	// 		Your are a very professional document summarization specialist.
+	// 		Please summarize the given document.
+	// `
+	// pdf := "gs://cloud-samples-data/generative-ai/pdf/2403.05530.pdf"
+	// location := "us-central1"
+	// modelName := "gemini-1.5-pro-preview-0409"
+	ctx := context.Background()
+
+	client, err := genai.NewClient(ctx, projectID, location)
+	if err != nil {
+		return fmt.Errorf("unable to create client: %v", err)
+	}
+	defer client.Close()
+
+	model := client.GenerativeModel(modelName)
+	model.SetTemperature(0.4)
+
+	part := genai.FileData{
+		MIMEType: "application/pdf",
+		FileURI:  pdf,
+	}
+
+	res, err := model.GenerateContent(ctx, part, genai.Text(prompt))
+	if err != nil {
+		return fmt.Errorf("unable to generate contents: %v", err)
```

Comment:
blocking: This and other fmt.Errorf instances, please wrap errors with %w.

Reply:
Awesome! I wasn't very familiar with `%w`, now I see it makes perfect sense